x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/MlMappingsUpgradeIT.java

```diff
@@ -105,8 +120,40 @@ private void assertUpgradedMappings() throws Exception {
             assertNotNull(propertiesLevel);
             // TODO: as the years go by, the field we assert on here should be changed
             // to the most recent field we've added that is NOT of type "keyword"
-            Map<String, Object> fieldLevel = (Map<String, Object>) propertiesLevel.get("multi_bucket_impact");
-            assertEquals(Collections.singletonMap("type", "double"), fieldLevel);
+            Map<String, Object> objectLevel = (Map<String, Object>) propertiesLevel.get("model_size_stats");
+            assertNotNull(objectLevel);
+            Map<String, Object> propertiesLevel2 = (Map<String, Object>) objectLevel.get("properties");
+            assertNotNull(propertiesLevel2);
+            Map<String, Object> fieldLevel = (Map<String, Object>) propertiesLevel2.get("peak_model_bytes");
+            assertEquals(Collections.singletonMap("type", "long"), fieldLevel);
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertUpgradedConfigMappings() throws Exception {
+
+        assertBusy(() -> {
+            Request getMappings = new Request("GET", ".ml-config/_mappings");
+            Response response = client().performRequest(getMappings);
+
+            Map<String, Object> responseLevel = entityAsMap(response);
+            assertNotNull(responseLevel);
+            Map<String, Object> indexLevel = (Map<String, Object>) responseLevel.get(".ml-config");
```

Comment:
The same here, could `XContentMapValues::extractValue` be used instead of this sequence of `get`s and casts?